src/System.Management.Automation/engine/regex.cs

```diff
@@ -965,16 +974,22 @@ public PatternPositionsVisitor(int lengthOfPattern)
 
                 _lengthOfPattern = lengthOfPattern;
 
-                _isPatternPositionVisitedMarker = new int[lengthOfPattern + 1];
-                for (int i = 0; i < _isPatternPositionVisitedMarker.Length; i++)
+                _isPatternPositionVisitedMarker = ArrayPool<int>.Shared.Rent(_lengthOfPattern + 1);
+                for (int i = 0; i <= _lengthOfPattern; i++)
                 {
                     _isPatternPositionVisitedMarker[i] = -1;
```

Comment:
Style comment - What about Array.Fill()?

Reply:
`Array.Fill` does the same with additional check. Also CrossGen doesn't handle generic types over value types. I think we should keep the current implementation.